pkg/repositories/mongo/player/get_players_test.go

```diff
@@ -0,0 +1,92 @@
+package player_test
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/jairogloz/go-l/pkg/domain"
+	"github.com/jairogloz/go-l/pkg/repositories/mongo"
+	"github.com/jairogloz/go-l/pkg/repositories/mongo/player"
+	"github.com/stretchr/testify/assert"
+	"github.com/testcontainers/testcontainers-go/modules/mongodb"
+)
+
+func TestService_GetPlayersByTeamID(t *testing.T) {
+	testTable := map[string]struct {
+		teamID        string
+		assertionFunc func(subTest *testing.T, p []*domain.Player, err error)
+	}{
+		"should return players by team id": {
+			teamID: "team1",
+			assertionFunc: func(subTest *testing.T, p []*domain.Player, err error) {
+				assert.Nil(subTest, err)
+				assert.Equal(subTest, 1, len(p))
+			},
+		},
+		"should return empty players list when team id not found": {
+			teamID: "team2",
+			assertionFunc: func(subTest *testing.T, p []*domain.Player, err error) {
+				assert.Nil(subTest, err)
+				assert.Equal(subTest, 0, len(p))
+			},
+		},
+	}
+
+	db, err := startMongoDB(context.Background())
+	if err != nil {
+		t.Fatalf("error starting mongodb container: %v", err)
+	}
+	defer db.Container.Terminate(context.Background()) // nolint: errcheck
+	repo, err := setupRepository(db)
+	if err != nil {
+		t.Fatalf("error setting up repository: %v", err)
+	}
+
+	for testName, test := range testTable {
+		t.Run(testName, func(subTest *testing.T) {
+			players, err := repo.GetPlayersByTeamID(test.teamID)
+			test.assertionFunc(subTest, players, err)
+		})
+	}
+}
+
+// setupRepository will return a player repository instance or an error
+// It will create an index in the collection for testing purposes
+// It will also create a player document in the collection for testing purposes
+func setupRepository(db *mongodb.MongoDBContainer) (*player.Repository, error) {
+	connString, err := db.ConnectionString(context.Background())
+	if err != nil {
+		return nil, err
+	}
+	client, err := mongo.ConnectClient(connString)
+	if err != nil {
+		return nil, err
+
+	}
+	playerRepo := &player.Repository{
+		Client:     client,
+		Collection: client.Database("go-l").Collection("players"),
+	}
+	if err = playerRepo.CreateIndexes(); err != nil {
+		return nil, err
+	}
+	if err := playerRepo.Insert(&domain.Player{
+		TeamInfo: &domain.TeamInfo{
+			TeamID: "team1",
+		},
+	}); err != nil {
+		return nil, err
+
+	}
```

Comment:
Entiendo que este insert cumple dos propósitos:
1.- Tener un player en la colección para poder luego consultarlo desde las pruebas
2.- Validar que la base de datos fue creada exitosamente. Aunque esto igual con la creación del index ya hay certeza. También se podría hacer un Ping() a la base de datos para garantizar que ya hay conexión.

Respecto al número 1, qué opinas sobre mover la inserción del player a la prueba en sí, antes del for, para que la función `setupRepository` se mantuviera independiente de cualquier prueba y pudiera ser usada desde cualquier prueba.